lib/ansible/modules/cloud/vmware/vsphere_guest.py

```diff
@@ -519,7 +521,11 @@ def add_nic(module, s, nfmor, config, devices, nic_type="vmxnet3", network_name=
             msg="Error adding nic backing to vm spec. No network type of:"
             " %s" % (network_type))
 
-    nic_ctlr.set_element_addressType("generated")
+    if mac_address == "GENERATED":
```

Comment:
This is also not documented in the module. If you are adding functionality and don't document it, what's the use ?